### DISCUSSION TITLE
Accessing `Select.Value` value

### DISCUSSION THREAD

**Discussion by ScreamZ** [radix-ui/primitives#2881] [Help]
Hi,

I would like to access `Select.Value` current value to apply some data transformation only for display.

```tsx
<Select.Value asChild placeholder={placeholder}>
    <MyComp />     
</Select.Value>
<Select.Icon>▼</Select.Icon>
```

I've tried this with MyComp but doesn't seems to work.

`MyComp` is using children, and truncate The text to 3 characters + ellipsis (...)

Any idea ?

Thanks

**Comment by allisonking** [comment]
This [example in the docs](https://www.radix-ui.com/primitives/docs/components/select#controlling-the-value-displayed-in-the-trigger) might help. In your case, you likely need to make your field "controlled" and then pass the field's value directly to `<MyComp />`
